sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/IndistinguishableSimilarElementsCheck.java

```diff
@@ -81,7 +81,7 @@ private void raiseViolationOnDuplicateLandmarkRole(List<TagNode> nodes) {
     Map<String, List<TagNode>> matched = nodes.stream().collect(Collectors.groupingBy(node -> node.getAttribute("ROLE").toUpperCase(Locale.ROOT)));
     for (List<TagNode> matches : matched.values()) {
       List<TagNode> labeless = matches.stream().filter(match -> !hasAriaLabel(match)).collect(Collectors.toList());
-      if (labeless.size() > 1) {
+      if (labeless.size() > 1 || (!labeless.isEmpty() && labeless.size() != matches.size())) {
```

Comment:
Note that `matches.size() > 1` condition would yield the same result. I let you decide whether you prefer to keep as is or not.

Reply:
Well spotted! I will change that.